src/core/server/elasticsearch/elasticsearch_config.test.ts

```diff
@@ -80,6 +81,45 @@ test('#hosts accepts both string and array of strings', () => {
   expect(configValue.hosts).toEqual(['http://some.host:1234', 'https://some.another.host']);
 });
 
+describe('#maxSockets', () => {
+  test('accepts positive numeric values', () => {
+    const configValue = new ElasticsearchConfig(config.schema.validate({ maxSockets: 512 }));
+    expect(configValue.maxSockets).toEqual(512);
+  });
+
+  test('throws if it does not contain a numeric value', () => {
+    expect(() => {
+      config.schema.validate({ maxSockets: 'foo' });
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"[maxSockets]: expected value of type [number] but got [string]"`
+    );
+
+    expect(() => {
+      config.schema.validate({ maxSockets: true });
+    }).toThrowErrorMatchingInlineSnapshot(
+      `"[maxSockets]: expected value of type [number] but got [boolean]"`
+    );
+  });
+
+  test('throws if it does not contain a valid numeric value', () => {
+    expect(() => {
+      config.schema.validate({ maxSockets: -1 });
+    }).toThrowErrorMatchingInlineSnapshot(
+      '"[maxSockets]: Value must be equal to or greater than [1]."'
+    );
+
+    expect(() => {
+      config.schema.validate({ maxSockets: 0 });
+    }).toThrowErrorMatchingInlineSnapshot(
+      '"[maxSockets]: Value must be equal to or greater than [1]."'
+    );
+
+    expect(() => {
+      config.schema.validate({ maxSockets: Infinity });
+    }).toThrowErrorMatchingInlineSnapshot('"[maxSockets]: \\"maxSockets\\" cannot be infinity"');
```

Comment:
Hum, I would have expected that to work, but I guess `Infinity` is not a valid `schema.number` value, surprising. 

If we want to support passing Infinity from the config file (as `.inf` if I trust the yaml spec), we may want to use

```ts
maxSockets: schema.oneOf(schema.number(...), schema.literal(Infinity))
```

That's not strictly necessary though.

Reply:
Yep, I thought `Infinity` would be a valid number too, but then the tests proved me wrong.